### PR TITLE
[NUI] TextField/TextEditor: Add SelectedTextStart, SelectedTextEnd pr…

### PR DESCRIPTION
**Pull request by Ali-Alzyoud**
TextField/TextEditor: Add SelectedTextStart, SelectedTextEnd

### Description of Change ###
Add properties for:
SelectedTextStart
SelectedTextEnd

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs [hunk 1]
```diff
@@ -239,6 +239,12 @@ internal static partial class TextEditor
             //manual pinvoke for text-editor ScrollStateChangedSignal
             [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_Dali_TextEditor_ScrollStateChangedSignal")]
             public static extern global::System.IntPtr TextEditor_ScrollStateChangedSignal(global::System.Runtime.InteropServices.HandleRef jarg1);
+
+            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_TextEditor_Property_SELECTED_TEXT_START_get")]
+            public static extern int TextEditor_Property_SELECTED_TEXT_START_get();
+
+            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_TextEditor_Property_SELECTED_TEXT_END_get")]
+            public static extern int TextEditor_Property_SELECTED_TEXT_END_get();
         }
```
Comment (huiyueun) [inline src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs:248]: https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/244453/
Please check the api name used in C # binder.

File: src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs [hunk 2]
```diff
@@ -239,6 +239,12 @@ internal static partial class TextEditor
             //manual pinvoke for text-editor ScrollStateChangedSignal
             [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_Dali_TextEditor_ScrollStateChangedSignal")]
             public static extern global::System.IntPtr TextEditor_ScrollStateChangedSignal(global::System.Runtime.InteropServices.HandleRef jarg1);
+
+            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_TextEditor_Property_SELECTED_TEXT_START_get")]
+            public static extern int TextEditor_Property_SELECTED_TEXT_START_get();
+
+            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_TextEditor_Property_SELECTED_TEXT_END_get")]
+            public static extern int TextEditor_Property_SELECTED_TEXT_END_get();
```
Comment (huiyueun) [inline src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs:247]: Please check the api name used in C # binder.
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/244453/

File: src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs [hunk 3]
```diff
@@ -239,6 +239,12 @@ internal static partial class TextEditor
             //manual pinvoke for text-editor ScrollStateChangedSignal
             [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_Dali_TextEditor_ScrollStateChangedSignal")]
             public static extern global::System.IntPtr TextEditor_ScrollStateChangedSignal(global::System.Runtime.InteropServices.HandleRef jarg1);
+
```
Comment (huiyueun) [inline src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs:242]: https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/244453/
Please check the api name used in C # binder.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 4]
```diff
@@ -983,6 +983,48 @@ public bool EnableSelection
...
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// This will be public opened in tizen_6.0 after ACR done, Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int SelectedTextEnd
+        {
+            get
+            {
+                int temp;
+                GetProperty(TextField.Property.SELECTED_TEXT_END).Get(out temp);
```
Comment (huiyueun) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1018]: I think it is correct to modify TextField to TextEditor//

**Comment by Ali-Alzyoud** [comment]
@huiyueun 
requested changes had been done.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs (hunk 4 above)
Comment (Seoyeon2Kim) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1018]: Hello, @Ali-Alzyoud , I think this is not resolved yet. Could you check it again?

**Review by AdunFang — APPROVED** [review]
It is OK.

**Review by rabbitfor — CHANGES_REQUESTED** [review]
@Ali-Alzyoud Please check Huiyu's comments and update the patch. Thank you.

**Review by rabbitfor — APPROVED** [review]
LGTM

**Review by Seoyeon2Kim — APPROVED** [review]
Looks good to me.

File: src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs (hunk 3 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs:242]: EntryPoint = "CSharp_TextEditor_Property_SELECTED_TEXT_START_get"

File: src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs [hunk 5]
```diff
@@ -239,6 +239,12 @@ internal static partial class TextEditor
...
+            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_Dali_TextEditor_Property_SELECTED_TEXT_START_get")]
+            public static extern int TextEditor_Property_SELECTED_TEXT_START_get();
+
+            [global::System.Runtime.InteropServices.DllImport(NDalicPINVOKE.Lib, EntryPoint = "CSharp_Dali_TextEditor_Property_SELECTED_TEXT_END_get")]
```
Comment (rabbitfor) [inline src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs:246]: EntryPoint = "CSharp_TextEditor_Property_SELECTED_TEXT_END_get"

Comment (Ali-Alzyoud) [reply src/Tizen.NUI/src/internal/Interop/Interop.TextEditor.cs:246]: @rabbitfor 
the function name is the same as in this patch
https://review.tizen.org/gerrit/gitweb?p=platform/core/uifw/dali-csharp-binder.git;a=blob;f=dali-csharp-binder/src/text-editor.cpp;h=2f30b82d354de9cb3aadd1e1465fa35f09d36876;hb=refs/heads/devel/master#l573

**Pull request merged**